app/src/services/queueSvc.js

```diff
@@ -149,11 +185,49 @@ class QueueService {
         await this.dataService.updateMessageSendResult(job.data.client, job.data.messageId, sendResult);
       } catch (e) {
         log.error(`Error sending message from queue: client = ${job.data.client}, messageId = ${job.data.messageId}. ${e.message}`);
-        log.error(JSON.stringify(e, null, 2));
+        log.error(utils.prettyStringify(e));
+      }
+    }
+  }
+
+  /**
+   * @function removeJob
+   * Attempts to remove the job from the queue.
+   *
+   * @param {string} client - the authorized party / client
+   * @param {object} jobId - the job id of the desired message
+   * @throws ClientMismatchError if job client does not match `client`
+   * @throws DataIntegrityError if job data is in an inconsistent state
+   * @throws UncancellableError if job state must is not 'delayed'
+   * @returns {boolean} True if successful, false if job was not found
+   */
+  async removeJob(client, jobId) {
+    const job = await this.queue.getJob(jobId);
+
+    if (job && job.data && job.data.client && job.data.messageId) {
+      // Job found with proper structure
+      const jobState = await job.getState();
```

Comment:
Unfortunately job.remove() has no such constraints or notions about whether a job is in progress or can't be removed - it just drops it from the queue at that moment. We are relegated to having to use job.getState() as it is the only exposed way of determining where the job's status is within Bull/Redis.

As for checking the job state, I changed the conditional to run the await getState to be inline in order to prevent getState from 1) being invoked and waited on unnecessarily (first two errors don't care about the state) and 2) we were only using that variable there anyways. This should tighten up things a bit.

Reply:
Cool, that can cut down on the number of times we call getState.
Please put in awaits on line 222/223.  We do have an order of operations and we should do the best we can to ensure it happens.  See comment above about connection pools and counting on db transaction locks.  Not a great idea when we can add 2 words of code.  Concern here would be that both 222 and 223 lines end up writing to the message table.  Let's just enforce an order of operations.

Reply:
I agree with restoring await on line 222 as there is another DB call that happens in 223 later that would benefit from waiting for 222 to finish first. I've also retrofitted that to the other queueListeners that follow the same pattern.
However, I don't agree with needing to put in an await for 223 (the updateContent call) because all later invocations do not actually need the resule of that updateContent.